wdl2cwl/tests/wdl_files/hisat2.wdl

```diff
@@ -0,0 +1,114 @@
+version 1.0
+
+# Copyright (c) 2017 Leiden University Medical Center
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+task Hisat2 {
+    input {
+        File inputR1
+        File? inputR2
+        Array[File]+ indexFiles
+        String outputBam
+        String sample
+        String library
+        String readgroup
+        String platform = "illumina"
+        Boolean downstreamTranscriptomeAssembly = true
+        #String summaryFilePath = basename(outputBam, ".bam") + ".summary.txt"
+        Int sortMemoryPerThreadGb = 2
+        Int compressionLevel = 1
+
+        Int? sortThreads
+
+        Int threads = 4
+        Int? memoryGb
+        #Int timeMinutes = 1 + ceil(size([inputR1, inputR2], "G") * 180 / threads)
+        # quay.io/biocontainers/mulled-v2-a97e90b3b802d1da3d6958e0867610c718cb5eb1
+        # is a combination of hisat2 and samtools hisat2=2.2.0 & samtools=1.10.
+        String dockerImage = "quay.io/biocontainers/mulled-v2-a97e90b3b802d1da3d6958e0867610c718cb5eb1:2880dd9d8ad0a7b221d4eacda9a818e92983128d-0"
+    }
+
+    # Samtools sort may block the pipe while it is writing data to disk.
+    # This can lead to cpu underutilization.
+    # 1 thread if threads is 1. For 2-4 threads 2 sort threads. 3 sort threads for 5-8 threads.
+    Int estimatedSortThreads = if threads == 1 then 1 else 1 + ceil(threads / 4.0)
+    Int totalSortThreads = select_first([sortThreads, estimatedSortThreads])
+    Int estimatedMemoryGb = 1 + ceil(size(indexFiles, "G") * 1.2) + sortMemoryPerThreadGb * totalSortThreads
+
+    command {
+        set -e -o pipefail
+        mkdir -p "$(dirname ~{outputBam})"
+        hisat2 \
+        -p ~{threads} \
+        
```

Comment:
Was there originally a blank line here? This doesn't seem right..

Reply:
No. I removed that line (```-x ~{sub(indexFiles[0], "\.[0-9]\.ht2", "")} \```) from the command because it contains the sub() function which is not yet implemented. I have to write a js expression for that which I thought could go to a separate PR. Shall I implement it in this PR itself?

Reply:
Got it. For now you could use a simpler expression that does not use `sub` like `-x ~{indexFiles[0]}` with a note that it won't be valid command line for `hisat2`. Later when you implement `sub` you can use the original expression and remove the note.